src/components/MediatedRequestsActivities/components/MediatedRequestsFilters/components/MediatedRequestLevelFilter/MediatedRequestLevelFilter.js

```diff
@@ -29,19 +51,34 @@ const MediatedRequestLevelFilter = () => {
         data-testid={MEDIATED_REQUEST_FILTER_TYPES.MEDIATED_REQUEST_LEVELS}
         header={FilterAccordionHeader}
         label={formatMessage({ id:'ui-requests-mediated.filters.mediatedRequestLevel.accordionTitle' })}
-        name={MEDIATED_REQUEST_FILTER_TYPES.MEDIATED_REQUEST_LEVELS}
+        name={filterPath}
         separator={false}
-        onClearFilter={() => {}}
+        displayClearButton={!isEmpty(activeValues)}
+        onClearFilter={() => clearGroup(filterPath)}
```

Comment:
Could you create separate function out of `return` statement?
Your approach can lead to unnecessary rerenders - https://marcoheine.com/blog/7-easy-ways-to-improve-your-react-performance-part-2

Reply:
Fixed